scheduler/plugin/plugin_test.go

```diff
@@ -41,19 +87,39 @@ func TestFilter(t *testing.T) {
 			expectedCode: framework.Success,
 		},
 		{
-			name:         "filtered: bad-node",
+			name:            "filtered: bad-node",
+			pod:             testdata.PodSmall,
+			node:            st.MakeNode().Name("bad-node").Obj(),
+			expectedCode:    framework.Unschedulable,
+			expectedMessage: "good-node != bad-node",
+		},
+		{
+			name:         "filtered: panic",
+			guestPath:    testdata.PathTestPanicOnFilter,
 			pod:          testdata.PodSmall,
-			node:         st.MakeNode().Name("bad-node").Obj(),
-			expectedCode: framework.Unschedulable,
+			node:         testdata.NodeSmall,
+			expectedCode: framework.Error,
+			expectedMessage: `wasm: filter error: panic!
+wasm error: unreachable
```

Comment:
I used manual webassembly (%.wat) because it is easier to reach error conditions and has stable stack traces. The wasm binaries are tiny so we can create a bunch of these if we need to.